app/src/main/java/ch/epfl/qedit/model/MatrixFormat.java

```diff
@@ -31,6 +35,36 @@ public MatrixFormat(
         this.hintString = hint();
     }
 
+    public static MatrixFormat parse(String format) {
+        /** Match format: 'matrixNxM' where N and M are [0-9]+ */
+        if (Pattern.compile("^matrix(\\d+)x(\\d+)$").matcher(format).find()) {
+            /** Extract the row and column size */
```

Comment:
It feels a little weak using strings. Shouldn't we use something better ? It will fail with the string 
"\s\s\s\smatrix 2x2\s\s\s". Where \s denotes a blank space.

Why does the regex isn't : 
^matrix(\d+)x(\d+)$
?

Reply:
let's support that indeed

Reply:
(\d+) does not compile on my machine